vault/resource_mfa_duo.go

```diff
@@ -81,55 +85,60 @@ func mfaDuoWrite(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[DEBUG] Creating mfaDuo %s in Vault", name)
 	_, err := client.Logical().Write(mfaDuoPath(name), data)
 	if err != nil {
-		return fmt.Errorf("error writing to Vault: %s", err)
+		return diag.Errorf("error writing to Vault: %s", err)
 	}
 
-	return mfaDuoRead(d, meta)
+	return mfaDuoRead(ctx, d, meta)
 }
 
-func mfaDuoDelete(d *schema.ResourceData, meta interface{}) error {
+func mfaDuoDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
-	name := d.Get("name").(string)
+	name := d.Get(consts.FieldName).(string)
 
 	log.Printf("[DEBUG] Deleting mfaDuo %s from Vault", mfaDuoPath(name))
 
 	_, err := client.Logical().Delete(mfaDuoPath(name))
 	if err != nil {
-		return fmt.Errorf("error deleting from Vault: %s", err)
+		return diag.Errorf("error deleting from Vault: %s", err)
 	}
 
 	return nil
 }
 
-func mfaDuoRead(d *schema.ResourceData, meta interface{}) error {
+func mfaDuoRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
-	name := d.Get("name").(string)
+	name := d.Id()
 
 	resp, err := client.Logical().Read(mfaDuoPath(name))
```

Comment:
We should probably be handling the case where `resp` is `nil`. In that case we would return nil to denote that resource does not exist in Vault.

Reply:
Good call! Added this check on [L121](https://github.com/hashicorp/terraform-provider-vault/pull/1669/files#diff-90462eca7e943acf538bf9f4b84b87b5588cde9d4cde65947b87c8f1e24683d7R121)